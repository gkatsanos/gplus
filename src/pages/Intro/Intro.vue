<style src="./intro.scss" lang="scss"></style>

<template>
  <div class="intro full-height flex flex-distribute">
    <div class="title-subtitle">
      <h2 class="subtitle">
        <template v-for="(character, index) in headlineSplit">
          <br v-if="character === '*'" :key="index" class="newline" />
          <span v-else-if="character === ' '" :key="index" class="space">
            {{ character }}
          </span>
          <span v-else :key="index" class="character">
            {{ character }}
          </span>
        </template>
      </h2>
    </div>
    <div class="skills">
      SCSS & JS architecture
      <span class="highlight">//</span>
      Vue, React, Node, Express
      <span class="highlight">//</span>
      webpack & scaffolding
      <br />
      performance optimization
      <span class="highlight">//</span>
      training & mentoring
    </div>
    <div class="social-links flex" role="navigation">
      <a rel="nofollow" href="https://github.com/gkatsanos/" target="_blank">
        <i class="icon-github"></i>
        github
      </a>
      <a
        rel="nofollow"
        href="http://stackoverflow.com/users/458060/george-katsanos"
        target="_blank"
      >
        <i class="icon-stackoverflow"></i>
        stackoverflow
      </a>
      <a rel="nofollow" href="https://www.linkedin.com/in/georgekatsanos" target="_blank">
        <i class="icon-linkedin-square"></i>
        linkedin
      </a>
      <a rel="nofollow" href="https://medium.com/@gplusgr" target="_blank">
        <i class="icon-medium"></i>
        medium
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: "Intro",
  data() {
    return {
      headline: "HUMAN-CENTERED*DESIGN & TECHNOLOGY",
    };
  },
  computed: {
    headlineSplit() {
      return this.headline.split("");
    },
  },
};
</script>
