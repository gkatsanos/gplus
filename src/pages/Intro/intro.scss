@import '../../sass/variables';
@import '../../sass/mixins/all';

.intro {
  h2.subtitle {
    text-align: center;
    max-width: 700px;
    font-weight: 500;
    font-size: $font-normal;
    @include mq('tablet') {
      font-size: 30px;
      margin-top: 42px;
    }
  }

  .character {
    @include mq('tablet') {
      width: 40px;
      height: 40px;
    }
    width: 15px;
    height: 15px;
    display: inline-block;
    vertical-align: middle;
    text-align: center;
    margin-bottom: 30px;
  }

  .space {
    display: inline-block;
    width: 15px;
    @include mq('tablet') {
      width: 20px;
    }
  }

  .skills {
    font-size: $font-medium;
    text-align: center;
  }
}

// Animation specific styling
.subtitle {
  @include loopchildren(0, 45, 0.1, 0.03, '.character');
}

.character {
  transition: opacity .1s ease-out;
}

.skills,
.social-links {
  transition: opacity .5s ease-in;
}

.fade-enter-active {
  .skills {
    transition-delay: 0.2s;
  }

  .social-links {
    transition-delay: 0.3s;
  }
}

.fade-leave-active {
  transition: opacity .5s ease-out;
}

.fade-leave-to {
  .title-subtitle {
    opacity: 0;
  }

  .skills,
  .social-links {
    opacity: 0;
  }
}

.fade-enter {
  .character {
    opacity: 0;
  }

  .skills,
  .social-links {
    opacity: 0;
  }
}
