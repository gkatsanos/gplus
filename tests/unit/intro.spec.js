import { shallowMount } from '@vue/test-utils';
import Intro from '@/pages/Intro/Intro.vue';

describe('Intro', () => {
  it('renders content properly', () => {
    const wrapper = shallowMount(Intro);
    expect(wrapper).toMatchSnapshot();
  });
});
