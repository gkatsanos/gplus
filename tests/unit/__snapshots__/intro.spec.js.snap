// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders content properly 1`] = `
<div class="intro flex flex-distribute">
  <div class="title-subtitle">
    <h2 class="subtitle"><span class="character">
              H
            </span><span class="character">
              U
            </span><span class="character">
              M
            </span><span class="character">
              A
            </span><span class="character">
              N
            </span><span class="character">
              -
            </span><span class="character">
              C
            </span><span class="character">
              E
            </span><span class="character">
              N
            </span><span class="character">
              T
            </span><span class="character">
              E
            </span><span class="character">
              R
            </span><span class="character">
              E
            </span><span class="character">
              D
            </span><br class="newline"><span class="character">
              D
            </span><span class="character">
              E
            </span><span class="character">
              S
            </span><span class="character">
              I
            </span><span class="character">
              G
            </span><span class="character">
              N
            </span><span class="space">

            </span><span class="character">
              &amp;
            </span><span class="space">

            </span><span class="character">
              T
            </span><span class="character">
              E
            </span><span class="character">
              C
            </span><span class="character">
              H
            </span><span class="character">
              N
            </span><span class="character">
              O
            </span><span class="character">
              L
            </span><span class="character">
              O
            </span><span class="character">
              G
            </span><span class="character">
              Y
            </span></h2>
  </div>
  <div class="skills">
    SCSS &amp; JS architecture
    <span class="highlight">//</span>
    Vue, React, Node, Express
    <span class="highlight">//</span>
    webpack &amp; scaffolding
    <br>
    performance optimization
    <span class="highlight">//</span>
    training &amp; mentoring
  </div>
  <div role="navigation" class="social-links"><a rel="nofollow" href="https://github.com/gkatsanos/" target="_blank"><i class="icon-github"></i>
      github
    </a> <a rel="nofollow" href="http://stackoverflow.com/users/458060/george-katsanos" target="_blank"><i class="icon-stackoverflow"></i>
      stackoverflow
    </a> <a rel="nofollow" href="https://www.linkedin.com/in/georgekatsanos" target="_blank"><i class="icon-linkedin-square"></i>
      linkedin
    </a> <a rel="nofollow" href="https://medium.com/@gplusgr" target="_blank"><i class="icon-medium"></i>
      medium
    </a></div>
</div>
`;
